### DISCUSSION TITLE
How to select all in Input widget?

### DISCUSSION THREAD

**Discussion by Tim-Cao** [Textualize/textual#3172] [Q&A] [answered]
I tried to use mouse and ctrl+a to select all text in Input widget. They didn't work.
![image](https://github.com/Textualize/textual/assets/52661397/2826a152-e2b3-4f3c-8ace-267b92b20561)

**Comment by davep** [comment, marked as answer]
`Input` has no concept of a "selection". 

You can see all the bindings for `Input` [in the Textual documentation for the `Input` widget](https://textual.textualize.io/widgets/input/#textual.widgets._input.Input.BINDINGS). <kbd>Ctrl</kbd>+<kbd>a</kbd> moves the cursor to the start of the input.

**Reply by Tim-Cao** [reply]
Is there a faster way to clear typed content?

**Reply by davep** [reply]
See the bindings linked to above, there are actions bound to keys for deleting all text to the left or the right of the cursor, for example.

**Reply by Tim-Cao** [reply]
I see. Many thanks
